graphula-core/src/Graphula.hs

```diff
@@ -445,69 +454,124 @@ type GraphulaNode m a =
   , EntityKeyGen a
   )
 
-{-|
-  Generate and edit a value with data dependencies. This leverages
-  'HasDependencies' to insert the specified data in the generated value. All
-  dependency data is inserted after editing operations.
+-- | Options for generating an individual node
+--
+-- @'NodeOptions'@ can be created and combined with the Monoidal
+-- operations @'(<>)'@ and @'mempty'@
+--
+-- @
+-- a1 <- root @A mempty
+-- a2 <- root @A $ primaryKey $ AKey 1
+-- a3 <- root @A $ edit $ \a -> a { someField = True }
+-- a4 <- root @A $ suchThat $ (== True) . someField
+-- a5 <- root @A $ primaryKey (AKey 2) <> edit (\a -> a { someField = True })
+-- a6 <- root @A $ primaryKey (AKey 3) <> do edit $ \a -> a { someField = True }
+-- @
+--
+data NodeOptions a = NodeOptions
+  { nodeOptionsEdit :: Endo (Maybe a)
```

Comment:
To avoid having to pass a redundant `Just a` to the starting function this could be kleisli composition for `<>`.